suite/tinytables/src/main/java/dk/alexandra/fresco/suite/tinytables/prepro/TinyTablesPreproResourcePool.java

```diff
@@ -2,62 +2,104 @@
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
-import java.util.Random;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class TinyTablesPreproResourcePool extends ResourcePoolImpl {
 
-  private static final Logger logger = LoggerFactory.getLogger(TinyTablesPreproResourcePool.class);
+  private static final Logger LOGGER = LoggerFactory.getLogger(TinyTablesPreproResourcePool.class);
+  private static final int OT_BATCH_SIZE = 16000;
 
-  private final Random random;
+  private final Drng random;
   private final List<TinyTablesPreproANDProtocol> unprocessedAnds;
   private final TinyTablesStorage storage;
   private final File tinyTablesFile;
+  private final TinyTablesOt baseOt;
+  private final RotList rotList;
+  private final CoinTossing ct;
+  private final OtExtensionResourcePool otExtRes;
   private TinyTablesTripleProvider tinyTablesTripleProvider;
 
   /**
    * Creates an instance of the default implementation of a resource pool. This contains the basic
    * resources needed within FRESCO.
    *
    * @param myId The ID of the MPC party.
+   * @param baseOt OT functionality for the base OTs
+   * @param random Secure randomness generator
    * @param noOfPlayers The amount of parties within the MPC computation.
    * @param tinyTablesFile file for data
    */
-  public TinyTablesPreproResourcePool(int myId, int noOfPlayers,
-      File tinyTablesFile) {
+  public TinyTablesPreproResourcePool(int myId, int noOfPlayers, TinyTablesOt baseOt, Drbg random,
+      int computationalSecurity, int statisticalSecurity, File tinyTablesFile) {
     super(myId, noOfPlayers);
-    this.random = new SecureRandom();
+    if (noOfPlayers != 2) {
+      throw new UnsupportedOperationException("TinyTable is only defined for 2 parties");
+    }
     this.unprocessedAnds = Collections.synchronizedList(new ArrayList<>());
     this.storage = new TinyTablesStorageImpl();
     this.tinyTablesFile = tinyTablesFile;
+    this.baseOt = baseOt;
+    this.rotList = new RotList(random, computationalSecurity);
+    this.ct = new CoinTossing(myId, Util.otherPlayerId(myId), random);
+    this.otExtRes = new OtExtensionResourcePoolImpl(myId, Util.otherPlayerId(myId),
+        computationalSecurity, statisticalSecurity, 1, random, ct, rotList);
+    this.random = new DrngImpl(random);
+  }
+
+  public Ot initializeOtExtension(Network network) {
+    baseOt.init(network);
+    int otherId = Util.otherPlayerId(getMyId());
+    // Execute random seed OTs
+    if (getMyId() < otherId) {
+      rotList.send(baseOt);
+      rotList.receive(baseOt);
+    } else {
+      rotList.receive(baseOt);
+      rotList.send(baseOt);
+    }
+    ct.initialize(network);
+    // Setup the OT extension
+    return new BristolOtFactory(new RotFactory(otExtRes, network), otExtRes, network,
+        OT_BATCH_SIZE);
   }
 
-  public Random getSecureRandom() {
+  public Drng getSecureRandom() {
```

Comment:
The name of this method is pretty confusing as it does not actually return anything of type `SecureRandom`. How about `getDrng()`?

Reply:
Changed the name,

Reply:
Well, it was the name of the public method `getSecureRandom` I was referring to :).

Reply:
I honestly though I did that, but you are right. I only change the internal name. Anyway now it is changed.